### PR TITLE
Fix UUID assignment and mqtt message spec

### LINKED ISSUE #15: uuid creation and spesification
There are three different uuid/id fields that need to be filled in the mqtt message
- id, this is a unique uuid for the mqtt message
- data\_id, some kind of identifier for the data in the message/data source for the message
- metadata\_id, an identifier for the metadata in the message
id should be a uuid with a namespaces. The namespace should reference the message creator.
It is suggested that data\_id and metadata_id just be the same as the message id.
I suggest the metadata\_id be an identifier we can use to get messages of the same type from the datastore.
i.e a hash of stationid and measurement type, making it easy to get the all of these measurements from the datastore.

### LINKED ISSUE #16: Update required fields in mqtt message.
Currently there are some shortcomings in the mandatory fields in the mqtt message spec. This should be updated to secure sufficient data and metadata will be embedded in the message for ingestion and rediscovery in the datastore.

### PR DESCRIPTION
**Pull request by Teddy-1000**
Closes #15 Closes #16

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 xarray==2023.7.0
-netCDF4==1.6.4
-numpy==1.25.2
-paho-mqtt==1.6.1
-jsonschema==4.19.0
+netCDF4<=1.6.4, <2.0.0
+numpy<=1.25.2, <2.0.0
+paho-mqtt<=1.6.1, <2.0.0
+jsonschema<=4.19.0, <5.0.0
+psycopg2>=2.9.7, <3.0.0
```
Comment (johtoblan) [inline requirements.txt]: What is the requirement for this one (psycopg2) again

Comment (johtoblan) [reply requirements.txt]: Still getting:
      Error: pg_config executable not found.

**Review by johtoblan — CHANGES_REQUESTED** [review]
Will do some more reviewing, but have a problem with the requirements again.

File: requirements.txt (hunk 1 above)
Comment (johtoblan) [reply requirements.txt]: annoying that the github job gets this right >:(

Comment (Teddy-1000) [reply requirements.txt]: We can just remove this requirement,. This is used for postgres connection when ingesting from kafka, and will not be needed for the ingestor.

Comment (Teddy-1000) [reply requirements.txt]: Removed psycopg2

File: src/esoh/ingest/main.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        if not input_type:
+            input_type = self.decide_input_type(message)
+
+        return self.build_message(message, input_type)
+
+    def decide_input_type(self):
+        pass
+
+    def build_messages(self, message: [str, object], input_type: str = None):
+        match input_type:
```
Comment (johtoblan) [inline src/esoh/ingest/main.py:32]: Why match case, isn't if sufficient and or better (one less indent)?

File: src/esoh/ingest/messages.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import xarray as xr
+
+import uuid
+
+from esoh.ingest.netCDF.extract_metadata_netcdf import build_all_json_payloads_from_netCDF
+
+
+def build_message(file: [object], input_type: str, uuid_prefix: str):
+    match input_type:
```
Comment (johtoblan) [inline src/esoh/ingest/messages.py:11]: same here

File: src/esoh/ingest/main.py (hunk 2 above)
Comment (Teddy-1000) [reply src/esoh/ingest/main.py:32]: The reason I chose to use this here is for ease of adding more filetypes at a later stage, and then making it easier to read when this list gets longer. I do agree that when there is only one filetype implemented there is no need to use match-case

File: src/esoh/ingest/messages.py (hunk 3 above)
Comment (Teddy-1000) [reply src/esoh/ingest/messages.py:11]: The reason I chose to use this here is for ease of adding more filetypes at a later stage, and then making it easier to read when this list gets longer. I do agree that when there is only one filetype implemented there is no need to use match-case

**Review by johtoblan — APPROVED** [review]
LGTM

**Pull request merged**